improver/wxcode/weather_symbols.py

```diff
@@ -198,7 +199,7 @@ def check_input_cubes(self, cubes):
                 # Check cube and threshold coordinate names match according to
                 # expected convention.  If not, add to exception dictionary.
                 if (
-                    get_threshold_coord_name_from_probability_name(diagnostic)
```

Comment:
Don't think this should have changed - the comment may not be correct but the code checks for the expected convention.

Reply:
Actually I was confused about this. I changed this line because otherwise the test `Test_construct_extract_constraint.text_zero_threshold` fails: the constraint has the wrong variable name and doesn't match the test cube. (I have changed that test to use the lightning diagnostic because lightning is the only one of the test cubes that contains a zero threshold. Previously the test was just checking that the string representing the constraint was as expected, using a diagnostic where the cube doesn't contain a zero threshold.)

So maybe the test cubes have not been set up to have the same names as they would have in production data?

Reply:
I made an attempt at fixing this, let me know if I've misunderstood. I changed the code that sets up the test cubes so that it will set the threshold coordinate name to be the variable name with "_in_vicinity" removed.

Now after changing `weather_symbols.py` back to use `get_threshold_coord_name_from_probability_name`, all the tests pass.

Reply:
You are right - I have just put up a PR to fix this and get rid of the "if / else" clause in weather symbols here.  I will ask you to rebase on that when it's done as that will make the testing of this change more robust.  But the conflicts should only be minor.